### PR TITLE
Departmental defense as the Found Fathers intended

### PR DESCRIPTION
**Pull request by Scrambledeggs00**
# Document the changes in your pull request

Adds in the maintenance musket and two cartridges for it: welding fuel cartridges and black powder cartridges. An easy to craft weapon, it is clunky to use, needing 5 seconds while not moving to reload, inaccurate and, if using the welding fuel cartridges, has negative AP. It also creates a cool smoke cloud :]

Cartridge stats:

welding fuel cartridge: 
damage = 25
AP = -30
spread = 40

black powder cartridge:
damage = 30
AP = 0
spread = 20

# Why is this good for the game?

Gives people without access to actual weapons an option which is easier to craft than the other ranged makeshift options.

# Testing

Crafting:
image
image

Musket in action:
https://www.youtube.com/watch?v=21SQyncZgwM

# Spriting

Mob:
MaintMusketHands

Object:
MaintMusketObject

# Wiki Documentation

The rifle does 5 damage in melee, takes 5 seconds of not moving to reload, and can have a bayonet attached to it. It is crafted with 1 pipe, 1 metal sheet, 1 rifle stock, and 5 package wrapping and needs a screwdriver, welder, and wrench for tools.

The welding fuel cartridge has the stats stated above. It is crafted with 1 cloth sheet and 10 units of welding fuel and needs a screwdriver for tools.

The black powder cartridge has the stats stated above. It is crafted with 1 cloth sheet and 10 units of black powder and needs a screwdriver for tools.

# Changelog

:cl:  
rscadd: Adds in the maintenance musket, welding fuel cartridge, and black powder cartridge. 
soundadd: Adds a firing and cocking sound for the musket.
imageadd: Adds sprites for the musket and cartridges.
/:cl:

### CODE REVIEW

**Comment by MommaDragon** [comment]
Okay, but can it get even better with improvised weapons knowledge?

**Comment by Scrambledeggs00** [comment]
> Okay, but can it get even better with improvised weapons knowledge?

I had originally planned on doing that, but I couldn't figure out how. Might come back to do it if I can think of how to do so.

**Comment by AMyriad** [comment]
image

**Comment by Addust** [comment]
MAKE READY

(amazing)

**Comment by Mqiib** [comment]
God bless spess america

**Review by Mqiib — APPROVED** [review]
Basically just slightly tweaked versions of the improv shotgun, save for the in-hands which are good enough.

File: code/modules/projectiles/guns/ballistic/launchers.dm [hunk 1]
```diff
@@ -148,3 +148,51 @@
...
+		to_chat(user, span_notice("You start reloading the [src]."))
+		user.visible_message(span_warning("[user] starts reloading the [src]!"))
+		reloading_active = TRUE
+		if(!do_after(user, 5 SECONDS, user))
+			reloading_active = FALSE
+			user.balloon_alert(user, "You were interupted!")
+			return
+		..()
+		chamber_round()
+		reloading_active = FALSE
```
Comment (Moltijoe) [inline code/modules/projectiles/guns/ballistic/launchers.dm]: there are so many things wrong with this part that it's hard for me to explain

File: code/modules/projectiles/guns/ballistic/launchers.dm [hunk 2]
```diff
@@ -148,3 +148,51 @@
...
+			return
+		else if(reloading_active == TRUE)
+			to_chat(user, span_warning("You're already reloading it!"))
+			return
+		to_chat(user, span_notice("You start reloading the [src]."))
+		user.visible_message(span_warning("[user] starts reloading the [src]!"))
+		reloading_active = TRUE
+		if(!do_after(user, 5 SECONDS, user))
+			reloading_active = FALSE
+			user.balloon_alert(user, "You were interupted!")
```
Comment (Moltijoe) [inline code/modules/projectiles/guns/ballistic/launchers.dm]:
```suggestion
			user.balloon_alert(user, "You were interrupted!")
```

File: code/modules/projectiles/guns/ballistic/launchers.dm [hunk 3]
```diff
@@ -148,3 +148,51 @@
...
+/obj/item/gun/ballistic/maintMusket/afterattack()
+	. = ..()
+	magazine.get_round(FALSE)
+
+/obj/item/gun/ballistic/maintMusket/attackby(obj/item/A, mob/user, params)
+	
+	if(istype(A, /obj/item/ammo_casing/caseless/cartridge) || istype(A, /obj/item/kitchen/knife))
+		if(istype(A, /obj/item/kitchen/knife)) //Attach bayonet
+			..()
+			return
```
Comment (Moltijoe) [inline code/modules/projectiles/guns/ballistic/launchers.dm]:
```suggestion
			return ..()
```

**Review by Moltijoe — CHANGES_REQUESTED** [review]
try looking at how other guns do attackby

File: code/modules/projectiles/guns/ballistic/launchers.dm (hunk 1 above)
Comment (Scrambledeggs00) [reply code/modules/projectiles/guns/ballistic/launchers.dm]: I made some changes. Did I miss anything?

File: code/modules/projectiles/guns/ballistic/launchers.dm [hunk 4]
```diff
@@ -148,3 +148,50 @@
 	. = ..()
 	playsound(loc, "sparks", 75, 1, -1)
 	do_sparks(8, 3, usr)
+
+/obj/item/gun/ballistic/maintMusket
```
Comment (Moltijoe) [inline code/modules/projectiles/guns/ballistic/launchers.dm]: don't use capitals in typepaths, use an underscore if you must

File: code/modules/projectiles/guns/ballistic/launchers.dm [hunk 5]
```diff
@@ -148,3 +148,50 @@
...
+	name = "maintenance musket"
+	desc = "A weapon with notoriously poor accuracy; it makes up for this by being quick and easy to smash together"
```
Comment (Moltijoe) [inline code/modules/projectiles/guns/ballistic/launchers.dm]:
```suggestion
	desc = "A weapon with notoriously poor accuracy; it makes up for this by being quick and easy to smash together."
```

File: code/modules/projectiles/guns/ballistic/launchers.dm [hunk 6]
```diff
@@ -148,3 +148,50 @@
...
+	icon_state = "maintMusket"
+	item_state = "maintMusket"
```
Comment (Moltijoe) [inline code/modules/projectiles/guns/ballistic/launchers.dm]: same as with type path, use underscores if you need to split words

File: code/modules/projectiles/guns/ballistic/launchers.dm [hunk 7]
```diff
@@ -148,3 +148,50 @@
...
+	internal_magazine = TRUE
+	empty_indicator = FALSE
+	can_bayonet = TRUE
+	knife_x_offset = 27
+	knife_y_offset = 13
+	can_suppress = FALSE
+	force = 5
+	cartridge_wording = "cartridge"
+	fire_sound = 'sound/weapons/musketShot.ogg'
+	load_sound = 'sound/weapons/musketCock.ogg'
```
Comment (Moltijoe) [inline code/modules/projectiles/guns/ballistic/launchers.dm]: these too

File: code/modules/projectiles/projectile/special/rocket.dm [hunk 8]
```diff
@@ -93,3 +93,15 @@
...
 	damage = 25
+
+/obj/projectile/bullet/cartridge
+	name = "cartridge"
+	desc = "A small metal ball fired from a musket."
+	damage = 25
+	armour_penetration = -30 //Cheap ammo means poor AP
+	spread = 40
+
+/obj/projectile/bullet/cartridge/blackPowder
```
Comment (Moltijoe) [inline code/modules/projectiles/projectile/special/rocket.dm]: here also

File: code/datums/components/crafting/weapons.dm [hunk 9]
```diff
@@ -453,6 +464,24 @@
 	time = 0.5 SECONDS
 	category = CAT_WEAPON_AMMO
 
+/datum/crafting_recipe/cartridgeWelder
```
Comment (Moltijoe) [inline code/datums/components/crafting/weapons.dm]: and here

File: code/datums/components/crafting/weapons.dm [hunk 10]
```diff
@@ -453,6 +464,24 @@
...
+	name = "Cartridge Welding Fuel" //Reversed so that they're together in the menu
+	result = /obj/item/ammo_casing/caseless/cartridge
+	reqs = list(/obj/item/stack/sheet/cloth = 1,
+				/datum/reagent/fuel = 10)
+	tool_behaviors = list(TOOL_SCREWDRIVER)
+	time = 2 SECONDS
+	category = CAT_WEAPON_AMMO
+
+/datum/crafting_recipe/cartridgeBP
```
Comment (Moltijoe) [inline code/datums/components/crafting/weapons.dm]: here

File: code/datums/components/crafting/weapons.dm [hunk 11]
```diff
@@ -188,6 +188,17 @@
 	time = 12
 	category = CAT_WEAPON_RANGED
 
+/datum/crafting_recipe/maintMusket
```
Comment (Moltijoe) [inline code/datums/components/crafting/weapons.dm]: here

File: code/modules/projectiles/ammunition/caseless/rocket.dm [hunk 12]
```diff
@@ -37,3 +37,17 @@
...
+
+/obj/item/ammo_casing/caseless/cartridge
+	name = "welding fuel cartridge"
+	desc = "A musket cartridge crafted by hand. This one uses welding fuel as its propellant."
+	caliber = CALIBER_MUSKET
+	icon_state = "cartridge"
+	projectile_type = /obj/projectile/bullet/cartridge
+	firing_effect_type = /obj/effect/particle_effect/fluid/smoke
+
+/obj/item/ammo_casing/caseless/cartridge/blackPowder
```
Comment (Moltijoe) [inline code/modules/projectiles/ammunition/caseless/rocket.dm]: here

File: code/modules/projectiles/guns/ballistic/launchers.dm [hunk 13]
```diff
@@ -148,3 +148,50 @@
...
+	
+	if(istype(A, /obj/item/ammo_casing/caseless/cartridge))
+		if(reloading_active == TRUE)
+			to_chat(user, span_warning("You're already reloading it!"))
+			return
+		if(magazine.stored_ammo.len > 0)
+			user.balloon_alert(user, "Already loaded!")
+			return
+		to_chat(user, span_notice("You start reloading the [src]."))
+		user.visible_message(span_warning("[user] starts reloading the [src]!"))
```
Comment (Moltijoe) [inline code/modules/projectiles/guns/ballistic/launchers.dm]: you can combine these together into one line, look at the visible_message proc to learn more

**Review by Moltijoe — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/ballistic/launchers.dm [hunk 14]
```diff
@@ -148,3 +148,49 @@
...
+	var/reloading_active = FALSE
+
+/obj/item/gun/ballistic/maint_musket/afterattack()
+	. = ..()
+	magazine.get_round(FALSE)
+
+/obj/item/gun/ballistic/maint_musket/attackby(obj/item/A, mob/user, params)
+	
+	if(istype(A, /obj/item/ammo_casing/caseless/cartridge))
+		if(reloading_active == TRUE)
```
Comment (Moltijoe) [inline code/modules/projectiles/guns/ballistic/launchers.dm]:
```suggestion
		if(reloading_active)
```
reloading_active is already a boolean

File: code/modules/projectiles/guns/ballistic/launchers.dm [hunk 15]
```diff
@@ -148,3 +148,49 @@
...
+			to_chat(user, span_warning("You're already reloading it!"))
+			return
+		if(magazine.stored_ammo.len > 0)
+			user.balloon_alert(user, "Already loaded!")
+			return
+		visible_message(span_warning("[user] starts reloading the [src]!"), span_notice("You start reloading the [src]."))
```
Comment (Moltijoe) [inline code/modules/projectiles/guns/ballistic/launchers.dm]:
```suggestion
		user.visible_message(span_warning("[user] starts reloading the [src]!"), span_notice("You start reloading the [src]."))
```

**Review by Moltijoe — CHANGES_REQUESTED** [review]

**Review by Moltijoe — APPROVED** [review]
pretty sure everything is good now

**Review by ToasterBiome — APPROVED** [review]
have at thee

**Pull request merged**